### ISSUE TITLE
Drop Pipenv, switch to e.g. pip-tools

### ISSUE THREAD

**Issue by Datamance** [lookit/lookit-api#562]
**TL;DR**: Consolidate dependency management, use pip-tools.

**Narrative [original]**

As a developer, I want dependency management to be as simple as possible. Use of Pipenv was originally introduced to make local development easier and allow automatically managed upgrades, however, two things have happened in the meantime:

1) Pipenv development has slowed to a grinding halt
2) We pin all of our dependencies now

This renders all the bells and whistles of pipenv fairly useless to us, and in fact it becomes a liability considering the fact that `pip install` in the CI pipeline could theoretically download a different set of transitive dependencies than are stipulated in the currently checked-in `Pipfile.lock`

We should have one source of truth, and `pip-tools` is perfectly suited for just that. We will use `pip-compile` to generate a static dependency specification that is not only checked into the codebase, but then actually used by our CI/CD pipeline. During local development, developers can use `pip-sync` to keep their virtual environments tightly in sync with the generated requirements.txt files, while fresh docker builds in the CI pipeline can just `pip install`, enjoying the benefits of a completely deterministic build.

**Acceptance Criteria**
- [ ] Finalize choice of package manager; consider advantages/disadvantages of poetry as well
- [ ] Pipfile and lockfile deleted
- [ ] Dependency versions installed are the same locally and by CI/CD pipeline
- [ ] Clear documentation of relevant setup; Invoke or Make task if needed
